pkg/podexecutor/pebinary.go

```diff
@@ -0,0 +1,305 @@
+// +build windows
+
+package podexecutor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/k3s-io/helm-controller/pkg/generated/controllers/helm.cattle.io"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/Microsoft/hcsshim"
+
+	"golang.org/x/sys/windows/registry"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	daemonconfig "github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/daemons/executor"
+	"github.com/rancher/rke2/pkg/bootstrap"
+	"github.com/rancher/rke2/pkg/images"
+	win "github.com/rancher/rke2/pkg/windows"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+)
+
+var (
+	ssldirs = []string{
+		"/etc/ssl/certs",
+		"/etc/pki/tls/certs",
+		"/etc/ca-certificates",
+		"/usr/local/share/ca-certificates",
+		"/usr/share/ca-certificates",
+	}
+)
+
+type PEBinary struct {
+	ManifestsDir    string
```

Comment:
Validate that you need these fields - ManifestsDir, Resolver, ImagesDir, etc seem like they're not necessary since things aren't being executing in pods or from images?